recipes/gtsam/all/conanfile.py

```diff
@@ -121,6 +121,8 @@ def config_options(self):
             del self.options.fPIC
             if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
                 raise ConanInvalidConfiguration ("GTSAM requirews MSVC >= 15")
+            if self.settings.compiler.runtime=="MT" or  self.settings.compiler.runtime=="MTd":
```

Comment:
I think you can fix this by doing 
```
tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "GtsamBuildTypes.cmake"), "/MDd", "")
tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "GtsamBuildTypes.cmake"), "/MD", "")
```

Reply:
(or/and move this to a patch)

Reply:
I'm not sure whats going on. For now, I disabled building entirely with MT or MTd and if this build succeeds I'll take a look to this.

Reply:
In that file, there are some definitions with `/MD` and `/MDd`.

Reply:
Erasing those definitions did not fix the problem but replacing them with `self.settings.compiler.runtime` seems to work.
Do you happen to know where those errors on Linux x86_64, Release, gcc 4.9, libstdc++ are coming from? I cant reproduce them on my end.
The log says
`cmake: relocation error: cmake: symbol _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEC1EPKcRKS3_, version GLIBCXX_3.4.21 not defined in file libstdc++.so.6 with link time reference
gtsam/4.0.2:`

Reply:
@ericriff Can you turn on verbose again? Maybe it will tell us at which step the error is occurring.

Reply:
I just launched a the linux builds on verbose mode.

Reply:
I think the problem lies with cmake.
cmake is probably built against a more recent libstdc++ then is installed on the builder containing gcc 4.9.
@uilianries  Can you check?

Reply:
I'm replicating the build L`inux x86_64, Release, gcc 4.9, libstdc++ . Options: gtsam:shared-True `on my end.
I checked and in both cases we're using the same cmake package (44fcf6b9a7fb86b2586303e3db40189d3b511830) so it should be the same, but somehow it compiles on my PC.